query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -36,14 +37,29 @@ pub async fn get_many_records(
 ) -> crate::Result<ManyRecords> {
     let field_names = selected_fields.db_names().map(String::from).collect();
     let idents: Vec<_> = selected_fields.types().collect();
-    let query = read::get_records(model, selected_fields.columns(), query_arguments);
-
-    let records = conn
-        .filter(query.into(), idents.as_slice())
-        .await?
-        .into_iter()
-        .map(Record::from)
-        .collect();
+    let mut records = Vec::new();
+
+    if query_arguments.can_batch() {
+        let batches = query_arguments.batched();
+        let mut futures = Vec::with_capacity(batches.len());
+
+        for args in batches.into_iter() {
+            let query = read::get_records(model, selected_fields.columns(), args);
+            futures.push(conn.filter(query.into(), idents.as_slice()));
+        }
+
+        for result in future::join_all(futures).await.into_iter() {
+            for item in result?.into_iter() {
+                records.push(Record::from(item))
+            }
+        }
```

Comment:
One optimization here would be using `FuturesUnordered`, that way you can start pushing to the records vec as soon as one future completes, instead of after all complete. Also probably premature optimization. edit: unless you need to preserve the ordering of futures, then `FuturesOrdered` where at least some of the time it will start work early.

Reply:
Will do.

Reply:
oh btw, will FuturesUnordered mess up the order of the items coming back?

Reply:
We can do this now when we order in our code.